Re-structure the packaging-style-guide

utkarsh2102 opened the pull request:
Re-structure the packaging-style-guide like other style-guides and move extra documentation to `docs/` in rubocop-packaging via https://github.com/utkarsh2102/rubocop-packaging/pull/20.

Also, use double_quotes instead of single_quotes.
And also add a TIP at the start about the guide hosted at https://packaging.rubystyle.guide.

Closes: #2

Signed-off-by: Utkarsh Gupta <<utkarsh@debian.org>>

utkarsh2102 committed:
Move "rationale" to rubocop-packaging docs/

Also use double_quotes instead of single_quotes.
And also add a TIP at the start about the guide
hosted at https://packaging.rubystyle.guide.

Closes: #2

Signed-off-by: Utkarsh Gupta <utkarsh@debian.org>

utkarsh2102 commented:
@bbatsov, looks better now? 😄

utkarsh2102 committed:
Trim "Examples" header

Signed-off-by: Utkarsh Gupta <utkarsh@debian.org>

bbatsov commented:
I think it's fine to have rationales in the style guide, the only think I suggested removing was references to any cops related to the guidelines.

utkarsh2102 committed:
Remove references of RuboCop::Packaging

Let this guide not be centered around the cops in the
rubocop-packaging extension and move its references
to docs/ in rubocop-packaging.

Signed-off-by: Utkarsh Gupta <utkarsh@debian.org>

utkarsh2102 committed:
Don't use references of cops

Remove references of cops that are provided by the
Packaging extension. And make this style guide more
generic and not specific & centered around the
Packaging extension.

Signed-off-by: Utkarsh Gupta <utkarsh@debian.org>

utkarsh2102 commented:
Hey,

> I think it's fine to have rationales in the style guide, the only think I suggested removing was references to any cops related to the guidelines.

Ah, I got it now! I've added two commits above doing so^ 😄 
Also, I moved rationale to docs now, so all the documentation is kept at one place :)

Let me know if things look in order here now?

bbatsov commented on README.adoc:
```diff
@@ -44,19 +47,6 @@ gem install rouge
 ----
 ====
 
-[TIP]
```
I think you should keep this tip, as we have similar in other style guides.

utkarsh2102 replied:
Okay, great!

bbatsov commented on README.adoc:
```diff
@@ -70,156 +60,77 @@ This guide is a work in progress - existing guidelines are constantly being impr
 guidelines are being added, and occasionally some guidelines would get removed.
 
 
-== Why Packaging Extension?
```
I think the rationale makes sense, but you should simply remove the references to RuboCop here.

utkarsh2102 replied:
Ah, fair enough, done! 😄

bbatsov commented on README.adoc:
```diff
@@ -70,156 +60,77 @@ This guide is a work in progress - existing guidelines are constantly being impr
-
-Avoid using `require_relative` with relative path to lib. Use `require` instead.
-
-==== Rationale [[require-relative-hardcoding-lib-rationale]]
```
You can keep rationales like this one IMO, but it's up to you.

utkarsh2102 replied:
Your advice is really valuable, thanks for that!
I've re-added them now! 😄

utkarsh2102 committed:
Re-add Rationale(s)

Thanks to @bbatsov's suggestions, it's good to keep rationales
but by simply removing the references to RuboCop.

Signed-off-by: Utkarsh Gupta <utkarsh@debian.org>

utkarsh2102 commented:
Okay, I hope I didn't miss out anything now 🙈 
Really thanks for your reviews, @bbatsov. Hopefully, this one should be last (I really hope!) 😄

utkarsh2102 commented:
Hey @bbatsov,

Thanks for your help so far!
I'd like to ask, would you be interested in using `rubocop-packaging` in RuboCop as well? 😄 
(it'd be great if so, it'll also help us to maintain it in Debian as well!)

utkarsh2102 commented:
@bbatsov, hi! 👋🏻 

Just a slight ping on this^
I'll be happy to raise a PR if so^? 😄
